### PR TITLE
system variable: add tidb_enable_rate_limit_action 

### PR DESCRIPTION
**Pull request by Yisaer**
Signed-off-by: Song Gao <disxiaofei@163.com>

### What is changed, added or deleted? (Required)
add `tidb_enable_rate_limit_action` into system variable.

This PR will be merged only after TiDB 4.0.9 is released.

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [x] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Comment by TomShawn** [comment]
@Yisaer Please involve a technical review~

**Comment by XuHuaiyu** [comment]
线下讨论了，会补充写内容

**Comment by Yisaer** [comment]
> 线下讨论了，会补充写内容

updated, PTAL @XuHuaiyu

**Review by XuHuaiyu — COMMENTED** [review]
LGTM

File: system-variables.md [hunk 1]
```diff
@@ -966,3 +966,10 @@ explain select * from t where age=5;
 +-------------------------------+------------+-----------+-----------------------------+-------------------------------+
 3 rows in set (0.00 sec)
 ```
+
+### `tidb_enable_rate_limit_action`
+
+- 作用域：SESSION | GLOBAL
+- 默认值：ON
+- 这个变量用于控制读数据算子是否开启动态内存控制功能。读数据算子默认情况下会最大启动 [tidb_disql_scan_concurrency](/system-variables.md#tidb_distsql_scan_concurrency) 个线程读取数据，当单条 sql 的内存使用每超过 [tidb_mem_quota_query](/system-variables.md#tidb_mem_quota_query) 一次，读数据算子会停下一个线程。
```
Comment (TomShawn) [inline system-variables.md:974]:
```suggestion
- 这个变量用于控制是否为读数据的算子开启动态内存控制功能。读数据的算子默认启用 [`tidb_disql_scan_concurrency`](/system-variables.md#tidb_distsql_scan_concurrency) 所允许的最大线程数来读取数据。当单条 SQL 语句的内存使用每超过 [`tidb_mem_quota_query`](/system-variables.md#tidb_mem_quota_query) 一次，读数据的算子会停止一个线程。
```

File: system-variables.md [hunk 2]
```diff
@@ -966,3 +966,10 @@ explain select * from t where age=5;
...
+- 当读数据算子的线程只剩 1 个的时候且单条 sql 内存使用继续超过 [tidb_mem_quota_query](/system-variables.md#tidb_mem_quota_query) 时候，该 sql 会触发其它的内存控制行为，如[落盘](/tidb-configuration-file.md#spilled-file-encryption-method)。
```
Comment (TomShawn) [inline system-variables.md:975]:
```suggestion
- 当读数据的算子只剩 1 个线程且当单条 SQL 语句的内存使用继续超过 [`tidb_mem_quota_query`](/system-variables.md#tidb_mem_quota_query) 时，该 SQL 语句会触发其它的内存控制行为，例如[落盘](/tidb-configuration-file.md#spilled-file-encryption-method)。
```

**Review by TomShawn — APPROVED** [review]
LGTM

**Comment by TomShawn** [comment]
@Yisaer Please resolve the conflict~

**Pull request merged**